shell/platform/android/io/flutter/embedding/android/AndroidTouchProcessor.java

```diff
@@ -220,6 +224,25 @@ private void addPointerForIndex(
     }
 
     int pointerKind = getPointerDeviceTypeForToolType(event.getToolType(pointerIndex));
+    // We use this in lieu of using event.getRawX and event.getRawY as we wish to support
+    // earlier versions than API level 29.
+    float viewToScreenCoords[] = {event.getX(pointerIndex), event.getY(pointerIndex)};
+    transformMatrix.mapPoints(viewToScreenCoords);
+    long buttons;
+    if (pointerKind == PointerDeviceKind.MOUSE) {
+      buttons = event.getButtonState() & 0x1F;
+      if (buttons == 0
```

Comment:
Add comment:

Some implementations translates trackpad scrolling into a mouse down-move-up event sequence with buttons: 0, such as ARC on a Chromebook. See https://github.com/flutter/engine/pull/11420, a legacy implementation that uses the same condition but converts differently.